src/vm/threads.cpp

```diff
@@ -9471,6 +9471,28 @@ void Thread::DeleteThreadStaticData()
     m_TLBTableSize = 0;
 }
 
+//+----------------------------------------------------------------------------
+//
+//  Method:     Thread::DeleteThreadStaticData   public
+//
+//  Synopsis:   Delete the static data for the given module. This is called
+//              when the AssemblyLoadContext unloads.
+//
+//
+//+----------------------------------------------------------------------------
+
+void Thread::DeleteThreadStaticData(ModuleIndex index)
+{
+    for (SIZE_T i = 0; i < m_TLBTableSize; ++i)
+    {
+        ThreadLocalBlock * pTLB = m_pTLBTable[i];
```

Comment:
What does guarantee that the thread is not growing these data structures when this is called from the thread destroying the module?

Reply:
The m_TLBTable contains one entry per each app domain. Since we have just one AppDomain, this array will never change. However, possible growth of the TLM table is really a problem, thank you for spotting it.

Reply:
Fixed by a new commit.